src/lin_diffEq_HHL.jl

```diff
@@ -19,106 +19,91 @@ export QuEuler, QuLeapfrog, QuAB2, QuAB3, QuAB4
 """
 
 """
-    alg_lin_diffeq
+    AlgQuDiff
     * step - step for multistep method
     * α - coefficients for xₙ
     * β - coefficent for xₙ'
 """
-struct alg_lin_diffeq
-  step::Int
-  α::Array
-  β::Array
-  alg_lin_diffeq(step,α,β) = new(step,α,β)
+
+struct AlgQuDiff
+    step::Int
+    α::Array
```

Comment:
This doesn't follow the performance tips, you should use parameteric type to allow the compiler to be able to specific your custom type, e.g

```julia
struct AlgQuDiff{T}
    step::Int
    α::Vector{T}
    β::Vector{T}
end
```

Reply:
And it would be more performant to use `SArray`/`MArray` in `StaticArrays` when `α` and `β` only contains a few elements.

Reply:
Yes, as much as possible, since this will make the interface more consistent with DE.jl

Reply:
Also, How do I make increase the accuracy of the HHL code. I am thinking increasing `n_reg ` and setting an appropriate `C-value` is all one can do. Keeping that in mind, the least error I get is of the order 0.01. My concern is that, if that is the case the higher order methods would not be effective in general.

Reply:
I'm not sure on this, any idea? @GiggleLiu

Reply:
> higher order methods would not be effective in general.

I don't know any version HHL algorithm can break the bottleneck of resolution `n_reg`. This review of HHL based algorithm may be helpful, but I don't have time to go through it for the moment
https://arxiv.org/abs/1802.08227

For phase estimation, I know an algorithm that can increase `n_reg` effectively without increasing the number of qubits (like in https://arxiv.org/abs/1507.08852). But it seems not work for HHL since here we must keep eigenvalues entangled.

I wonder how many number of qubits have you used as registers for readouts to achieve a precision of 0.01? And could you please explain why higher order methods are affected by poor resolution, are you using finite difference in solving differential equations?

Reply:
I used `n_reg` = 12.
My thinking was that, the global error in a higher order method would be overshadowed by the resolution error, so we would get a similar result even if were to take a low order method.